src/coreclr/jit/compiler.cpp

```diff
@@ -2297,6 +2297,13 @@ void Compiler::compSetProcessor()
             codeGen->GetEmitter()->SetUseEvexEncoding(true);
             // TODO-XArch-AVX512 : Revisit other flags to be set once avx512 instructions are added.
         }
+        if (canUseRex2Encoding() || DoJitStressRex2Encoding())
+        {
+            // TODO-Xarch-apx:
+            //   At this stage, since no machine will pass the CPUID check for APX, we need a special stress mode that
+            //   enables REX2 on incompatible platform, `DoJitStressRex2Encoding` is expected to be removed eventually.
```

Comment:
This is a mis-statement, we will keep this REX2 stress mode. The difference between REX2 stress and the existing EVEX stress is that now REX2 does not do the CPUID check, we will need to add it when we have the compatible machine. Or, we may make use of the `JitBypassAPXCheck` knob to skip the CPUID check.

Reply:
I think we can just rely on the AltJit and/or NAOT scenario for "bypassing" the CPUID check, seeing as we can't run the tests anyways.

There are the existing mechanisms for generating code for a CPU that doesn't match the host CPU.

Reply:
> I think we can just rely on the AltJit and/or NAOT scenario for "bypassing" the CPUID check, seeing as we can't run the tests anyways.

Thanks for the explanation! And could you elaborate more on this technique that I can "get" APX on non-APX machine?

If I follow the existing EVEX stress mode, inside `DoJitStressEvexEncoding`, it will return true only if Avx512/Avx10 is available, so `DoJitStressRex2Encoding` will require APX available, can you give me some instructions on how I can get the ISA check pass on non-APX machine? Thanks!

Reply:
Normally just using the `AltJit` will work, however it only works with mismatched VM today (like if you're on x64 Windows, then using the x64 Unix AltJit).

I've put up https://github.com/dotnet/runtime/pull/110320 which will allow you to get the same regardless of which AltJit is used.

You normally do `DOTNET_AltJit=*` and `DOTNET_AltJitName=clrjit_win_x64_x64.dll` (for example) to get AltJit for all methods. You can replace `*` with a specific method name and the alt jit with other variants, like `clr_unix_x64_x64.dll` to cover other variations